src/core/thread/mle.cpp

```diff
@@ -246,6 +252,7 @@ ThreadError Mle::Stop(void)
         mNetif.RemoveUnicastAddress(mLeaderAloc);
     }
 
+    mNetif.GetPendingDataset().StopDelayTimer();
```

Comment:
Maybe I'm missing something, but `thread stop` will cause the device to detach from the network and can then manage its Delay Timer based on its local pending dataset.  My previous point is that we should restart the delay timer when:

1. the device has attached to a network and received a Pending Dataset from its Parent.
2. the device has detached from the network on its own (i.e. due to loss of connectivity).

I understand that we currently need to stop the Delay Timer as part of `Mle::Stop()` and that is fine for now.  `Mle::Stop()` is only called as part of `thread stop`, so it's fine to stop the Delay Timer in there for now.  But we should also consider restarting the Delay Timer if the device has detached from the network on its own (i.e. due to losing connectivity with its Parent).  Remember that there are two different reasons for detaching from the network: (i) the device detaches due to calling `thread stop` and (ii) the device detaches due to loss of connectivity.  Right now, you are attempting to handle (i).  I think we can also handle (ii) properly.